### PR TITLE
fix bounding rectangles for RectangleGeometry

### PR DESCRIPTION
**Pull request by likangning93**
[on CesiumJS](https://cesiumjs.org/Cesium/Build/Apps/Sandcastle/#c=rVRtb9owEP4rFl8IEjgwqdK0pWgt7aZprSqxrdMkvjjJAR6OHdkXWjrx33fOCwTKqk1bPoDvfK/Pc761sGwt4QEsO2caHtgEnCwyfl/qgm5SihOjUUgNttt7O9Mzva69bk0Kihx/zjSjzxoUKI1mb9iQD2d6WxrXEVfaJCtTIEcrklWwc68izgudlK6ZWMEUEhR6oSBoIvaaFFWtHDRKlOC4hcys4UKpoIpzykakaVC7l1U20anMltp/iTE2lVogOLqsC99Vw+fWZFewsAAuGIxG1GOfvSp/B68r4YwPe/3DmBlFs1KofcCJUcby6fUVf5C4vFD5UgRDfnbs2EKzdrwVuORopiKVQrs9OHu/bXXc7sEIQ/ajcMiWYIHNjWW4hD0ENWBE5071AUwGaDdEa3sgpsf3v4P031H728Zb3fpOfCa9S09t/K95eAYR32n+kPPv1zc3d99epp2eiKKnRq5oC+h7ApcgF0tkmecxBuYASyYbSzQslS5XYnMyUpn6uJLLm4vJp9Nzs21eOBqjYuEXQ2qSIiP8+ALwWoE/Xm4+pkG3NinXwvE7F3muNpdSE4ALt3/v/SZu7+R2oAx3sQO7FjG9/5ZXt2G92+OuiF1iZQxBVXezPQIa2HuhCui1mTxcKbmwDt4rI3Bv7UupYaiqOnShKa33XjVJT8ZkX0xwNFeVSaffiRxuFIyrkO9klhuLrLAq4DxEyIgomqcwLpIVIE9c6RiFjVOUyjWT6fmsc7R7Zx2WKOEc3cwLpT7LJ5h1xlFI9gdu1JqH/G4N1o8EmSxH45tKyTmPQhKfe9WkkHlVd4SegHGEsUk34z2YEVpSpuNpzUYUktC+bkulRuq8QIabHCiNJUSpbJZJTdJg5DeAF8UjiY3kEHIvlkIqUAxiGiPSrD1bb3YLos9KxdecbEjdLTN1dy2crgDhEX0Ogo+ks5czHMRqtUpHgiGswKH/EivPYgXrLw)
[on localhost:8080](http://localhost:8080/Apps/Sandcastle/index.html#c=rVRtb9owEP4rFl8IEjgwqdK0pWgt7aZprSqxrdMkvjjJAR6OHdkXWjrx33fOCwTKqk1bPoDvfK/Pc761sGwt4QEsO2caHtgEnCwyfl/qgm5SihOjUUgNttt7O9Mzva69bk0Kihx/zjSjzxoUKI1mb9iQD2d6WxrXEVfaJCtTIEcrklWwc68izgudlK6ZWMEUEhR6oSBoIvaaFFWtHDRKlOC4hcys4UKpoIpzykakaVC7l1U20anMltp/iTE2lVogOLqsC99Vw+fWZFewsAAuGIxG1GOfvSp/B68r4YwPe/3DmBlFs1KofcCJUcby6fUVf5C4vFD5UgRDfnbs2EKzdrwVuORopiKVQrs9OHu/bXXc7sEIQ/ajcMiWYIHNjWW4hD0ENWBE5071AUwGaDdEa3sgpsf3v4P031H728Zb3fpOfCa9S09t/K95eAYR32n+kPPv1zc3d99epp2eiKKnRq5oC+h7ApcgF0tkmecxBuYASyYbSzQslS5XYnMyUpn6uJLLm4vJp9Nzs21eOBqjYuEXQ2qSIiP8+ALwWoE/Xm4+pkG3NinXwvE7F3muNpdSE4ALt3/v/SZu7+R2oAx3sQO7FjG9/5ZXt2G92+OuiF1iZQxBVXezPQIa2HuhCui1mTxcKbmwDt4rI3Bv7UupYaiqOnShKa33XjVJT8ZkX0xwNFeVSaffiRxuFIyrkO9klhuLrLAq4DxEyIgomqcwLpIVIE9c6RiFjVOUyjWT6fmsc7R7Zx2WKOEc3cwLpT7LJ5h1xlFI9gdu1JqH/G4N1o8EmSxH45tKyTmPQhKfe9WkkHlVd4SegHGEsUk34z2YEVpSpuNpzUYUktC+bkulRuq8QIabHCiNJUSpbJZJTdJg5DeAF8UjiY3kEHIvlkIqUAxiGiPSrD1bb3YLos9KxdecbEjdLTN1dy2crgDhEX0Ogo+ks5czHMRqtUpHgiGswKH/EivPYgXrLw)

### CODE REVIEW

**Comment by cesium-concierge** [comment]
Signed CLA is on file.

@likangning93, thanks for the pull request! Maintainers, we have a signed CLA from @likangning93, so you can review this at any time.

:warning: I noticed that [CHANGES.md](https://github.com/AnalyticalGraphicsInc/cesium/blob/master/CHANGES.md) has not been updated. If this change updates the public API in any way, fixes a bug, or makes any non-trivial update, please add a bullet point to `CHANGES.md` and comment on this pull request so we know it was updated. For more info, see the [Pull Request Guidelines]( https://github.com/AnalyticalGraphicsInc/cesium/blob/master/CONTRIBUTING.md#pull-request-guidelines).

---

__I am a bot who helps you make Cesium awesome! [Contributions to my configuration are welcome.](https://github.com/AnalyticalGraphicsInc/cesium/tree/master/.concierge)__

:earth_africa: :earth_americas: :earth_asia:

**Comment by hpinkos** [comment]
Looks good @likangning93!  I'm just going to wait until after the release to merge this in

File: Source/Core/RectangleGeometry.js [hunk 1]
```diff
@@ -562,36 +562,27 @@ define([
...
 
-        ellipsoid.cartesianArrayToCartographicArray(scratchRectanglePoints, scratchCartographicPoints);
+        var positions = scratchRectanglePoints;
+        RectangleGeometryLibrary.computePosition(options, 0, 0, positions[0], stScratch);
+        RectangleGeometryLibrary.computePosition(options, 0, width - 1, positions[1], stScratch);
+        RectangleGeometryLibrary.computePosition(options, height - 1, 0, positions[2], stScratch);
+        RectangleGeometryLibrary.computePosition(options, height - 1, width - 1, positions[3], stScratch);
 
-        return Rectangle.fromCartographicArray(scratchCartographicPoints);
+        return Rectangle.fromCartesianArray(positions);
```
Comment (hpinkos) [inline Source/Core/RectangleGeometry.js]: @likangning93 This is something I missed from your `EllipseGeometry` changes.  You need to pass in `rectangleGeometry._ellipsoid` here as the second argument.  Please make the same change in `EllipseGeometry` too

**Comment by hpinkos** [comment]
That's the last thing @likangning93 then this will be good to go

**Comment by likangning93** [comment]
@hpinkos updated!

**Comment by hpinkos** [comment]
Awesome, thanks @likangning93!

**Pull request merged**